cql3/functions/castas_fcts.cc

```diff
@@ -133,11 +134,11 @@ simple_date_native_type time_point_to_date(const db_clock::time_point& tp) {
 }
 
 db_clock::time_point date_to_time_point(const uint32_t date) {
-    const auto epoch = boost::posix_time::from_time_t(0);
-    const auto target_date = epoch + boost::gregorian::days(int64_t(date) - (1UL<<31));
-    boost::posix_time::time_duration duration = target_date - epoch;
-    const auto millis = std::chrono::milliseconds(duration.total_milliseconds());
-    return db_clock::time_point(std::chrono::duration_cast<db_clock::duration>(millis));
+    // "date" counts the number of days since the epoch, where the middle
+    // of the unsigned range, 2^31, signifies the epoch itself.
+    int64_t millis_since_epoch = (int64_t(date) - (1UL<<31)) * 24 * 60 * 60 * 1000;
+    return db_clock::time_point(std::chrono::duration_cast<db_clock::duration>(
+        std::chrono::milliseconds(millis_since_epoch)));
 }
```

Comment:
How sure are you that the change is correct? Do you know what boost::gregorian::days does?

Reply:
I'm not 100% sure, but 99% sure because I saw that the ASAN error reported was a simple multiplication by 86400000000, i.e., 24 hours in microseconds.
I also added in this PR (I forgot to mention it in the commit message) another test that tests timestamps like one day before and after the epoch and saw I got the right numbers.

But since you asked, I'll go look now at the gregorian::days() code and verify they don't have any crazy leap-second adjustment or julian dates or Dinosaur-age dates or anything like that that maybe my test would have missed -  and report here back.

Reply:
ok. btw I prefer writing number_of_milliseconds * 1ms instead of the huge cast expression.